main.go

```diff
@@ -25,7 +53,117 @@ var pongMessage = []byte(`{
 	]
 }`)
 
+type State struct {
+	db *sql.DB
+}
+
+var s *State
+
+func NewState(path string) (*State, error) {
+
+	db, err := sql.Open("postgres", path)
+	if err != nil {
+		return nil, err
+	}
+
+	err = db.Ping()
+	if err != nil {
+		return nil, err
+	}
+
+	s := &State{
+		db: db,
+	}
+	return s, nil
+}
+
+func cleanMessage(message []byte) *Msg {
+
+	var msg map[string]interface{}
+
+	if err := json.Unmarshal(message, &msg); err != nil {
+		log.Println(err)
+	}
+
+	m := &Msg{
+		msg: msg,
+	}
+
+	return m
+}
+
+func (m Msg) decodeMsg() Block {
+
+	// FORMAT of msg
+	// {
+	// 	"emit": [
+	// 	   "..",
+	// 	   {
+	// 		   "block": {
+	//				number : xxxxx
+	//				hash : xxxxx
+	//				...
+	//			}
+	// 	   }
+	// 	]
+	// }
+
+	decodedMsg := (m.msg["emit"]).([]interface{})[1].(map[string]interface{})["block"]
```

Comment:
Again, you are assuming that this message contains a block, but what happens when we add another message like this?
```
{
   "emit": {
     "..",
     {
         "something else": {
            "key": "val"
         }
     }
   }
}
```
You are either gonna have to write custom `Msg` object for each one which ends up messy or you could reuse the same Msg object like this:
```
func (m Msg) decodeMsg(field string, out interface{}) Block {
 	decodedMsg := (m.msg["emit"]).([]interface{})[1].(map[string]interface{})[field]

        rawMsg, _ := json.Marshal(decodedMsg)
	if err := json.Unmarshal(rawMsg, out); err != nil {
		panic(err)
        }
}
```
and then call the function like this:
```
msg := ..
var rawBlock Block
msg.decodeMsg("block", &rawBlock)
```

Reply:
Hey what should I put in the return type? If I put interface.. we will need marshall unmarshall it again..

Also, it checks if the block is present then only this function is called.

else if strings.Contains(string(message), "REORGS DETECTED") {

			m := cleanMessage(message)
			block := m.decodeMsg()
			s.WriteBlock(block, "reorgblocks")

		} else if strings.Contains(string(message), "block") {

			m := cleanMessage(message)
			block := m.decodeMsg()
			s.WriteBlock(block, "blocks")
}